server/src/main/kotlin/fi/oph/kitu/csvparsing/CsvConversionExtensions.kt

```diff
@@ -34,20 +34,47 @@ inline fun <reified T> String.asCsv(args: CsvArgs = CsvArgs()): List<T> {
     val csvMapper = getCsvMapper<T>()
     val schema = getSchema<T>(csvMapper, args)
 
-    return try {
-        csvMapper
-            .readerFor(T::class.java)
-            .with(schema)
-            .readValues<T>(this)
-            .readAll()
-    } catch (e: InvalidFormatException) {
+    // the lines are needed to read line by line in order to distinguish all erroneus lines
+    val errors = mutableListOf<CsvExportError>()
+    val data =
+        this
+            .split(args.lineSeparator)
+            .mapIndexed { index, line ->
+                try {
+                    csvMapper
+                        .readerFor(T::class.java)
+                        .with(schema)
+                        .readValue<T?>(line)
+                } catch (e: InvalidFormatException) {
+                    errors.add(CsvExportError(index, e))
+                    null
+                }
+            }.filterNotNull()
+
+    if (errors.isEmpty()) {
+        return data
+    }
+
+    // add all errors to log
+    errors.forEachIndexed { index, error ->
```

Comment:
Harmillisesti näyttää siltä että CsvArgs-parametria ei anneta sisään YkiServicestä, jolloin näitä virheitä ei kai lokiteta ikinä mihinkään. Pitäiskö tää muuttaa tässä samassa?

Reply:
korjattu (commit: https://github.com/Opetushallitus/koto-rekisteri/pull/542/commits/b0c6bc2f1709c6f94cb5b73498141798a9f2a132)